src/app/item-popup/ItemStat.tsx

```diff
@@ -176,3 +218,32 @@ function getModdedStatValue(item: DimItem, stat: DimStat) {
 export function isD1Stat(item: DimItem, _stat: DimStat): _stat is D1Stat {
   return item.isDestiny1();
 }
+
+function breakDownTotalValue(statValue: number, item: D2Item) {
+  let baseTotalValue = statValue;
+  let totalModsValue = 0;
+  let totalMasterworkValue = 0;
+
+  if (item.sockets && item.sockets.sockets) {
+    for (const socket of item.sockets.sockets) {
+      if (socket.plug && socket.plug.stats) {
+        const plugStats = socket.plug.stats;
+        armorStats.forEach((statHash) => {
+          const plugStatValue = plugStats[statHash];
+          if (plugStatValue) {
+            baseTotalValue -= plugStatValue;
+            totalModsValue += plugStatValue;
+          }
+        });
+      }
+    }
+  }
+
+  // In Armour 2.0 if an item is masterworked it gets a +2 to all 6 stat values
```

Comment:
One thing to check is whether masterworking adds a mod that has +2 in all stats, or whether it's something else.

Reply:
Hmmm that's true. Where would I likely find that info?
I don't think I have the materials in game to go do a class item atm.

Reply:
I'm not exactly sure - I haven't really played much yet. I'm just wondering if there's a filled socket that says "I'm a masterwork, here are my stats" vs. the masterwork being intrinsic.

Reply:
It looks to me that status is filled out by the plug.isMasterwork field (sockets.ts line 226). I will update it to use this but if you think it may be different in 2.0 maybe it would be safest to wait until we can get ahold of real data.

Alternatively looking at the bungie api, specifically [plugs](https://bungie-net.github.io/multi/schema_Destiny-Definitions-Items-DestinyItemPlugDefinition.html) and their [ui style](https://bungie-net.github.io/multi/schema_Destiny-PlugUiStyles.html) that field could potentially be used but using the isMasterworked would be nicer.